utils/EA/selection.py

```diff
@@ -0,0 +1,26 @@
+from random import sample
+from operator import itemgetter
+from .config import maximization, tour_size, elite_percentage
+
+# Parents Selection: tournament
+def tournament(population):
+    mate_pool = []
+    for i in range(len(population)):
+        winner = one_tour(population)
```

Comment:
does this always select the same phenotype

Reply:
I hope not :-D
The trick is inside the one_tour function. In each iteration it takes 3 random samples from the population and returns the genotype (phenotype would be the array of selected features) with the highest fitness score out of these 3 samples

Reply:
There exists also a Roulette Wheel selection, but this has a higher genetic pressure, whether tournament selection with tournament size 3 should lead to higher diversity. We can discuss this later if you want